src/main/java/de/tum/in/www1/artemis/config/MetricsBean.java

```diff
@@ -113,13 +165,231 @@ private void registerWebsocketMetrics() {
+
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+
+        var activeUsers = statisticsRepository.getActiveUsers(ZonedDateTime.now().minusDays(numberOfDaysToCountAsActive), ZonedDateTime.now());
+        var activeUserNames = activeUsers.stream().map(StatisticsEntry::getUsername).collect(Collectors.toCollection(ArrayList::new));
+
+        return exerciseRepository.countActiveStudentsInExercisesWithCurrentOrUpcomingReleaseDateWithinTimeRange(now, endDate, exerciseType.getExerciseClass(), activeUserNames);
+    }
+
+    private double getUpcomingEndingExamCount(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamsWithEndDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    private double getUpcomingEndingExamCountWithStudentMultiplier(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamUsersInExamsWithEndDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    private double getUpcomingStartingExamCount(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamsWithStartDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    private double getUpcomingStartingExamCountWithStudentMultiplier(int minutes) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return examRepository.countExamUsersInExamsWithStartDateGreaterThanEqualButLessOrEqualThan(now, endDate);
+    }
+
+    @Scheduled(fixedRate = 15 * 60 * 1000, initialDelay = 0) // Every 15 minutes
+    public void updatePublicArtemisMetrics() {
+        // The authorization object has to be set because this method is not called by a user but by the scheduler
+        SecurityUtils.setAuthorizationObject();
+
+        ZonedDateTime now = ZonedDateTime.now();
+
+        var activeUserPeriodsInDays = new Integer[] { 1, 7, 14, 30 };
+        activeUserMultiGauge.register(Stream.of(activeUserPeriodsInDays)
+                .map(periodInDays -> MultiGauge.Row.of(Tags.of("period", periodInDays.toString()), statisticsRepository.countActiveUsers(now.minusDays(periodInDays), now)))
+                .collect(Collectors.toCollection(ArrayList::new)), true);
+
+        var courses = courseRepository.findAll();
+        // We set the number of students once to prevent multiple queries for the same date
+        courses.forEach(course -> course.setNumberOfStudents((long) userRepository.getStudents(course).size()));
```

Comment:
Fetching user objects for all students in the course only to throw them away immediately because only the count is actually used.